packages/reactivity/src/arrayInstrumentations.ts

```diff
@@ -280,7 +280,12 @@ function reduce(
   let wrappedFn = fn
   if (arr !== self) {
     if (!isShallow(self)) {
+      let needReactive = args.length === 0
       wrappedFn = function (this: unknown, acc, item, index) {
+        if (needReactive) {
```

Comment:
```suggestion
if (args.length === 0 && index <= 1) {
  acc = toReactive(acc)
}
```

Reply:
This condition didn't account for scenario using `reduceRight`. In this case, it's necessary to check if `index >= arr.length - 2`. Therefore, I've modified it to the current implementation, which is simpler and doesn't need to access the array's length on each iteration.